.github/workflows/release-and-publish.yml

```diff
@@ -29,11 +34,30 @@ jobs:
       uses: actions/setup-go@v2
       with:
         go-version: 1.18.2
-    # Build Binaries
+    # Check if the newest tag already exists
+    - uses: mukunku/tag-exists-action@v1.0.0
+      id: check-tag-exists
+      with: 
+        tag:  "v${{ steps.changelog_reader.outputs.version }}"
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    # Build Binaries if this is a new release
     - name: Build Draft Release Binaries
       run: make build-all
-    # Create Release
+      if: ${{ steps.check-tag-exists.outputs.exists == 'false' }}
+    # If the tag already exists, we don't upload release artifacts
+    - name: Update Draft Release Body
+      if: ${{ steps.check-tag-exists.outputs.exists == 'true' }}
+      uses: softprops/action-gh-release@v1
+      env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+      with:
+        tag_name : "v${{ steps.changelog_reader.outputs.version }}"
+        name: "v${{ steps.changelog_reader.outputs.version }}"
+        body: ${{ steps.changelog_reader.outputs.changes }}
+    # Create Release with artifacts
     - name: Create Draft Release
```

Comment:
if the tag already exists, then we only update the content of the release body to match the changelog, so there is no `files` input. This is to avoid overriding the release artifacts when updating the release body